datarobot_batch_scoring/batch_scoring.py

```diff
@@ -117,105 +131,404 @@ def run_batch_predictions(base_url, base_headers, user, pwd,
+                        writer_status.value == b"I":
+                    ui.info("All requests done, waiting for writer")
+                    if writer_proc:
+                        writer_queue.put((WriterQueueMsg.SENTINEL, {}))
+                    if network_proc:
+                        network_queue.put(SENTINEL)
+                    aborting_phase = -1
+                    phase_start = time()
+
+            elif aborting_phase == -1:
+                procs = [shovel_proc, network_proc, writer_proc]
+                not_done = [a is False for a in [shovel_done,
+                                                 network_done,
+                                                 writer_done]]
+                if (procs == [None, None, None] and
+                        not_done == [False, False, False]):
+                    ui.info("all workers exited successfully")
+                    break
+                elif time() - phase_start > 30:
+                    ui.info("some of workers are still active, aborting")
+                    if writer_done != "ok":
+                        exit_code = 1
+                    aborting_phase = 1
+
+            elif aborting_phase == 1:
+                abort_flag.value = 1
+                aborting_phase = 2
+                phase_start = time()
+                ui.info("abort sequence started, waiting for workers exit")
+
+            elif aborting_phase == 2:
+                procs = [shovel_proc, network_proc, writer_proc]
+                if procs == [None, None, None]:
+                    ui.info("all workers exited")
+                    break
+                elif time() - phase_start > 10:
+                    for proc in procs:
+                        if proc and proc.is_alive():
+                            proc.terminate()
+
+                    aborting_phase = 3
+                    phase_start = time()
+
+            elif aborting_phase == 3:
+                procs = [shovel_proc, network_proc, writer_proc]
+                if procs == [None, None, None]:
+                    ui.info("all workers exited")
+                    break
+                elif time() - phase_start > 10:
+                    for proc in procs:
+                        if proc and proc.is_alive():
+                            os.kill(proc.pid, 9)
+                aborting_phase = 4
+
+            elif aborting_phase == 4:
+                procs = [shovel_proc, network_proc, writer_proc]
+                if procs == [None, None, None]:
+                    ui.info("all workers exited")
+                    break
+                elif time() - phase_start > 10:
+                    ui.error("some workers are not exited, ignoring")
+                    break
+
+        if shovel_done:
+            ui.info("Shovel is finished {}. Chunks produced: {}"
+                    " Resource usage: {}"
+                    "".format(shovel_done, s_produced, s_rusage))
```

Comment:
s_rusage is a dict so it prints in a random order (at least between runs). Can you format it so it displays in the same order?

Bonus points if you rename the keys to something more user friendly like `SystemTime`/`UserTime`/`Memory` instead of `{'utime': 52.673207, 'rss': 68588, 'stime': 1.631294}`